dotnet/src/webdriver/Proxy.cs

```diff
@@ -93,93 +95,90 @@ public Proxy()
         /// Initializes a new instance of the <see cref="Proxy"/> class with the given proxy settings.
         /// </summary>
         /// <param name="settings">A dictionary of settings to use with the proxy.</param>
+        /// <exception cref="ArgumentNullException">If <paramref name="settings"/> is <see langword="null"/>.</exception>
+        /// <exception cref="ArgumentException">If The "noProxy" value is a list with a <see langword="null"/> element.</exception>
         public Proxy(Dictionary<string, object> settings)
         {
             if (settings == null)
             {
                 throw new ArgumentNullException(nameof(settings), "settings dictionary cannot be null");
             }
 
-            if (settings.ContainsKey("proxyType") && settings["proxyType"] != null)
+            if (settings.TryGetValue("proxyType", out object? proxyTypeObj) && proxyTypeObj?.ToString() is string proxyType)
```

Comment:
Can you teach me what is `proxyTypeObj?.ToString() is string`? If operand is `null`, then `null is string` returns...

Reply:
I just folded two null checks into one:

```csharp
if (proxyTypeObj is not null)
{
    string? proxyType = proxyTypeObj.ToString();
    if (proxyType is not null)
    {
        ...
    }
}
```

This is part of the unfortunate situation that an untyped `object.ToString()` may return a null string. In fact, we can do that today with `new SessionId(null).ToString()`. We need to guard against that possibility here, since this method is `public`.

Reply:
In general is it right syntax: `null is string?` ?

Reply:
I mean this construction is not yet clear. Or you think it is normal and widely used?

Reply:
if a type is `string?` then `is string` will just be a null check. We can also do `proxyTypeObj?.ToString() is { } proxyType` if that's more clear.

Reply:
> you think it is normal and widely used?

It's used when people want to do this kind of thing: https://grep.app/search?q=%3F.ToString%28%29%20is%20string

Reply:
People like the `is { }` construction more it looks like https://grep.app/search?q=%3F.ToString%28%29%20is%20%7B

Reply:
Thanks, so it is about null checking rather than type checking. Please keep in mind that `if () {}` is better in terms of maintainability- I mean please don't be so aggressive to re-style existing code.

Reply:
I understand the concern with avoiding unnecessary changes. The changes we're talking about here are a direct result of nullability improvements. I'll try to avoid unnecessary style changes.